core/eth/utils.go

```diff
@@ -126,3 +128,27 @@ func bitmapToBytesArray(bitmap *big.Int) []byte {
 	}
 	return bytesArray
 }
+
+func isZeroValuedReservation(reservation paymentvault.IPaymentVaultReservation) bool {
+	return reservation.SymbolsPerSecond == 0 &&
+		reservation.StartTimestamp == 0 &&
+		reservation.EndTimestamp == 0 &&
+		len(reservation.QuorumNumbers) == 0 &&
+		len(reservation.QuorumSplits) == 0
+}
+
+// ConvertToActiveReservation converts a upstream binding data structure to local definition.
+// Returns an error if the input reservation is zero-valued.
+func ConvertToActiveReservation(reservation paymentvault.IPaymentVaultReservation) (*core.ActiveReservation, error) {
+	if isZeroValuedReservation(reservation) {
+		return nil, fmt.Errorf("reservation does not exist for given account")
```

Comment:
This error message is not accurate given this is a generic util method

Reply:
updated to say that `reservation is not a valid active reservation`